src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs

```diff
@@ -62,9 +62,11 @@ public class AzResourceTypeProvider : IResourceTypeProvider
             "identity",
             "managedByExtended",
             "tags",
+            "asserts",
         };
 
         public static readonly TypeSymbol Tags = new ObjectType(nameof(Tags), TypeSymbolValidationFlags.Default, Enumerable.Empty<TypeProperty>(), LanguageConstants.String, TypePropertyFlags.None);
+        public static readonly TypeSymbol Asserts = new ObjectType(nameof(Asserts), TypeSymbolValidationFlags.Default, Enumerable.Empty<TypeProperty>(), LanguageConstants.Bool, TypePropertyFlags.None);
```

Comment:
Do asserts support runtime function calls and runtime property accesses? If not, we may need to set `TypePropertyFlags.DeployTimeConstant` here.

Reply:
not yet.. I'm gonna set the flag here now then make changes once that happens